### PR TITLE
Sign up and approve accounts pages

### LINKED ISSUE #216: sign up page: 2 days

### LINKED ISSUE #217: sign up queue management (if required): 1 day

### PR DESCRIPTION
**Pull request by struan**
This adds pages to allow a user to request and account and an admin to approve those newly created accounts.

Fixes #217 
Fixes #216

One outstanding thing this doesn't do is assign a user to an organisation. I'm not totally sure this is required and it would add a chunk of friction in to the process as it would require an "assign organisation" dropdown for each user in the activation screen. But then if the user is from an org we've not added how does that get handled?

I don't think I'd want a user to pick an org at signup? Maybe it could have a combined select org or write in if not there, and then the activation person can confirm the org, but again have the problem of orgs not added.

### CODE REVIEW

**Review by zarino — CHANGES_REQUESTED** [review]
Generally looks good. A few thoughts:

- Signup page (well, all account-request-related pages) need to be accessible without being logged in, which isn’t currently the case.
- Reviewers might not be able to recognise people from their email address and organisation alone, so we probably want to collect the person’s name during signup, and expose that to the reviewer on `/activate_accounts`.
- Not related to this PR, but it’s confusing that the box you type your _email address_ into on the `/accounts/login` page is labelled "Username". We ask people for their "Email" on `/accounts/signup`, we should ask for the same on `/accounts/login`.
- If a user attempts to log in with their new details _before_ their account is approved, they get a generic "Please enter a correct username and password" error at `/accounts/login`. Is there any way we could detect that they’re trying to log into an un-approved account, and show a message like "Your account hasn’t been approved yet" instead?

**Review by zarino — CHANGES_REQUESTED** [review]

File: hub/templates/hub/accounts/activate_accounts.html [hunk 1]
```diff
@@ -31,6 +32,9 @@ <h1 class="mb-5">Activate accounts</h1>
                             {% bootstrap_field user_form.is_active show_help=False show_label=False wrapper_class="" %}
                             {% bootstrap_field user_form.id %}
                         </td>
+                        <td>
+                            {{ user_form.instance.first_name }} {{ user_form.instance.last_name }}
```
[the remark this answers is not in the document]
Comment (struan) [reply hub/templates/hub/accounts/activate_accounts.html]: done

**Comment by struan** [comment]
OK, I think I've resolved all the issues with this.

**Review by zarino — APPROVED** [review]
@struan this looks good to me! I’ve added a couple of commits to the end. If you’re happy with them, feel free to merge.

**Pull request merged**